### ISSUE TITLE
io.realm.internal.Table.finalize() timed out after 10 seconds

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#2496]
Copy from #1599 
People are still seeing finalizer timeouts on 0.88.2: http://crashes.to/s/360245fd72a
We should look into how hard it would be to get rid of finalizers completely IMO.

/cc @dalinaum

**Comment by GaborPeto** [comment]
If there are no finalizers how can I still get this exception in v1.2?
` java.util.concurrent.TimeoutException: io.realm.internal.Table.finalize() timed out after 10 seconds
       at java.util.regex.Matcher.closeImpl(Matcher.java)
       at java.util.regex.Matcher.finalize(Matcher.java:652)
       at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:187)
       at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:170)
       at java.lang.Thread.run(Thread.java:841)`

**Comment by beeender** [comment]
@GaborPeto the PR(#3144) to remove finalizer  has not been merged yet. Sorry for the exception you got.

**Comment by lorenzopolidori** [comment]
Does anybody know in which Realm release this fix will be? I can still see this crash using realm-java 2.2.1 on android.
```
Fatal Exception: java.util.concurrent.TimeoutException: io.realm.internal.Table.finalize() timed out after 10 seconds
       at io.realm.internal.Table.finalize(SourceFile:88)
       at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:206)
       at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:189)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by beeender** [comment]
@lorenzopolidori The fix has not been released yet. Currently only our snapshot release contains this fix. The fix will be released in `v2.2.2` soon.
